railgun/internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -218,6 +218,7 @@ func (p *clientgoCachedProxyResolver) startCacheServer() {
 			updateConfigSHA, err := p.kongUpdater(p.ctx, p.lastConfigSHA, p.cache, p.ingressClassName, p.deprecatedLogger, p.kongConfig, p.enableReverseSync)
 			if err != nil {
 				p.logger.Error(err, "could not update kong admin")
+				break
```

Comment:
Just swallow error silently ?

Reply:
The error is logged, but not acted on. Is that what you mean? This is an intentional (and historical) choice in behaviors that's consistent with KIC 1.x. the break added here was just to avoid making any changes to the `clientgoCachedProxyResolver` object that were unnecessary, it is a micro-optimization.

Reply:
@ccfishk and I talked about this in a zoom today, just to clarify: the problem here was that _technically_ there's no reason to assume the values provided back from a function which returns an error are valid. This `break` stops us from updating the object with what is essentially invalid data during an error case.